domain/secret/service/service.go

```diff
@@ -375,16 +316,37 @@ func (s *SecretService) UpdateUserSecret(ctx context.Context, uri *secrets.URI,
 		}
 	}
 
-	err := s.st.UpdateSecret(ctx, uri, p)
-	return errors.Annotatef(err, "updating user secret %q", uri.ID)
+	revisionID, err := s.uuidGenerator()
```

Comment:
We only want to do this if there's content to update.
Adding it to the upsert params gets rid of the messiness here.